src/metabase/app_db/connection_pool_setup.clj

```diff
@@ -46,24 +74,68 @@
   [^Class klass]
   (let [field (doto (.getDeclaredField com.mchange.v2.c3p0.C3P0Registry "classNamesToConnectionCustomizers")
                 (.setAccessible true))]
-
     (.put ^java.util.HashMap (.get field com.mchange.v2.c3p0.C3P0Registry)
           (.getName klass) (.newInstance klass))))
 
-(register-customizer! DbActivityTracker)
+(register-customizer! MetabaseConnectionCustomizer)
 
 (def ^:private application-db-connection-pool-props
-  "Options for c3p0 connection pool for the application DB. These are set in code instead of a properties file because
-  we use separate options for data warehouse DBs. See
-  https://www.mchange.com/projects/c3p0/#configuring_connection_testing for an overview of the options used
+  "Options for c3p0 connection pool for the application DB. These are set in code instead of the `c3p0.properties` file
+  because we use separate options for data warehouse DBs, and setting them in the properties file would affect both.
+  See https://www.mchange.com/projects/c3p0/#configuring_connection_testing for an overview of the options used
   below (jump to the 'Simple advice on Connection testing' section.)"
-  (merge
-   {"idleConnectionTestPeriod" 60
-    "connectionCustomizerClassName" (.getName DbActivityTracker)}
-   ;; only merge in `max-pool-size` if it's actually set, this way it doesn't override any things that may have been
-   ;; set in `c3p0.properties`
-   (when-let [max-pool-size (config/config-int :mb-application-db-max-connection-pool-size)]
-     {"maxPoolSize" max-pool-size})))
+  {;;
```

Comment:
Added documentation for all of the settings so we don't need to go open c3p0 dox to know what they do.